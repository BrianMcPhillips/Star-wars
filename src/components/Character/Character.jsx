import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name }) => (
  <>
    <h2>{name}</h2>
    <h1>Character</h1>
  </>
);

Character.propTypes = {
  name: PropTypes.string.isRequired
};

export default Character;
