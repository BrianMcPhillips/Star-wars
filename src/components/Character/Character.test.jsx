import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';
import { MemoryRouter } from 'react-router-dom';

describe('Character component', () => {
  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Character 
          name="Luke Skywalker"
          url="https://swapi.dev/api/people/1/"
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
