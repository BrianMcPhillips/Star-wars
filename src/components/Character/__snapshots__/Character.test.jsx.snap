// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <div>
    <h2>
      <a
        href="/detail/1/"
      >
        Luke Skywalker
      </a>
    </h2>
  </div>
</DocumentFragment>
`;
